plugin/discovery/find.go

```diff
@@ -0,0 +1,167 @@
+package discovery
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+const machineName = runtime.GOOS + "-" + runtime.GOARCH
```

Comment:
Not sure it matters at all, but use an underscore to match existing Go pkg directories? No sweat if you don't like the `_` for some reason ;)